Evaluate mathematical expression using eval

jnvance opened the pull request:
This adds the ability to evaluate expressions.

I'm adding some text to the description during the review which will make it to the final commit message in the case of a squash commit.

jnvance committed:
Evaluate mathematical expression using ast

jnvance37 commented on main.py:
```diff
@@ -0,0 +1,15 @@
+def evaluate_expression(expression):
+    try:
+        # Use eval to evaluate the expression
+        result = eval(expression)
```
Please do not use eval to evaluate user input. This is dangerous. Use ast.literal_eval instead.

jnvance37 reviewed (changes requested):
See changes requested.

jnvance37 commented on main.py:
```diff
@@ -0,0 +1,20 @@
+        result = evaluate_expression(expression)
+        print(f"Result: {result}")
+    except (ValueError, SyntaxError):
+        print("Invalid expression")
```
This should be clearly shown as an error.

jnvance37 commented on main.py:
```diff
@@ -0,0 +1,20 @@
+
+def evaluate_expression(expression):
+    try:
+        # Use ast.literal_eval to evaluate the expression safely
+        parsed_expression = ast.literal_eval(expression)
+        return parsed_expression
```
Add the ability to store variables as well.

jnvance committed:
Evaluate expressions and store them as variables

jnvance committed:
Raise all errors to get a full stack

I've added this commit so that there is a commit message

I will put some more details about the commit here.

We wanted to check how we can review commit messages(?)
